linux/DtaDevLinuxSata.cpp

```diff
@@ -487,7 +649,10 @@ void DtaDevLinuxSata::identify_SAS(OPAL_DiskInfo *disk_info)
     }
 
     // fill out disk info fields
-    safecopy(disk_info->serialNum, sizeof(disk_info->serialNum), resp->m_T10VendorId, sizeof(resp->m_T10VendorId));
+    if (!g_compat_bsd)
+        safecopy(disk_info->serialNum, sizeof(disk_info->serialNum), resp->m_T10VendorId, sizeof(resp->m_T10VendorId));
+    else
+        get_serial_from_vpd(fd, disk_info->serialNum); // We expect it to be successful since the SCSI INQUIRY command succeeded
```

Comment:
Device might have no serial number, there might be an I/O error, ...  The expectation is questionable.

Reply:
Thanks. Updated to fallback to `resp->m_T10VendorId` in case of error.

Reply:
I haven't looked, but I doubt FreeBSD used `m_T10VendorId` if it could not get a serial number.  Worth checking.

Reply:
Yeah, FreeBSD indeed does not doing that. Updated.